### ISSUE TITLE
Workflow RO-Crate: clarify spec on workflow types

### ISSUE THREAD

**Issue by simleo** [workflowhub-eu/about#46]
The current spec says:

The _Crate_ MUST contain a data entity of type `["File", "SoftwareSourceCode", "ComputationalWorkflow"]` as the _Main Workflow_.

This could be interpreted as saying that the types must be exactly `["File", "SoftwareSourceCode", "ComputationalWorkflow"]`, while WorkflowHub actually allows for more types to appear on the list. This is required by the [Provenance Run Crate](https://www.researchobject.org/workflow-run-crate/profiles/provenance_run_crate) profile (which adds `HowTo`), so I propose we change the spec as follows:

The _Crate_ MUST contain a _Main Workflow_ data entity whose types MUST include `File`, `SoftwareSourceCode` and `ComputationalWorkflow`.

**Comment by fbacall** [comment]
I missed this. I think this is a good idea.

I'm wondering what the best-practice way of changing this is now we have the versioned spec. Do you think it would be a breaking change to implementers? I guess possibly, if they are looking for the exact 3 terms to identify something as a main workflow...

**Comment by simleo** [comment]
> if they are looking for the exact 3 terms

Yes, it's potentially breaking in case anyone is doing such a strict check, however unlikely. So the profile's version should change to 2.0.

In any event, it's very important that WorkflowHub keeps allowing the extra types as it does now, even with the current spec, so that Provenance Run Crates can be uploaded to the registry.

**Comment by stain** [comment]
I would say it should bump to version 1.1, not 2.0, as anyone following 1.0 would still be in compliance even if we are now laxer.
